src/core/common/code_utils.hpp

```diff
@@ -172,4 +172,22 @@
  */
 static inline void IgnoreError(otError aError) { OT_UNUSED_VARIABLE(aError); }
 
+/**
+ * Executes the `statement` and performs action on success.
+ *
+ * Checks for the specified status, which is expected to commonly be successful, and executes action if
+ * status was successful.
+ *
+ * @param[in]  aStatus     A scalar status to be evaluated against zero (0).
+ *
+ */
+#define OnSuccess(...)                        \
+    do                                        \
+    {                                         \
+        if ((OT_FIRST_ARG(__VA_ARGS__)) == 0) \
+        {                                     \
+            OT_SECOND_ARG(__VA_ARGS__);       \
+        }                                     \
+    } while (false)
```

Comment:
I understand the idea of adding `OnSuccess()` to simplify the code (condensing an `if()` check to a single line) in `HandleSingleTlv()`, but I would recommend against adding this.

We've been selective about the `code_utils.hpp` macros and limited them to a few (`VerifyOrExit()` and `SuccessOrExit()`) to keep the OpenThread code easy to follow and understand for new developers and readers. They are primarily intended to abstract away `goto exit` and perform error guard checks.

Reply:
please feref to https://github.com/openthread/openthread/pull/10526#discussion_r1699973720